metricbeat/module/system/raid/raid_test.go

```diff
@@ -15,17 +15,61 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//+build darwin freebsd linux openbsd
+
 package raid
 
 import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
 	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
 )
 
+//MockData is a fake type for testing
+type MockData string
+
+//Close closes the connection to the /dev/md device
+func (dev MockData) Close() error {
+	return nil
+}
+
+//GetArrayInfo returns a struct describing the state of the RAID array.
+func (dev MockData) GetArrayInfo() (MDArrayInfo, error) {
+
+	dat := MDArrayInfo{
+		MajorVersion:  1,
+		MinorVersion:  2,
+		Ctime:         0,
+		Level:         0,
+		RAIDDisks:     1,
+		NrDisks:       1,
+		NotPersistent: 0,
+		Utime:         0,
+		State:         1,
+		ActiveDisks:   1,
+		WorkingDisks:  1,
+		FailedDisks:   0,
+		SpareDisks:    5,
+		ChunkSize:     0,
+	}
+	return dat, nil
+}
+
+func init() {
+	mb.Registry.MustAddMetricSet("system", "raid-test", New,
+		mb.WithHostParser(parse.EmptyHostParser),
+	)
```

Comment:
I think we don't need to register a new metricset.

Reply:
Ah, for some reason I just assumed that an `init()` function needed that.